src/test/java/com/xceptance/neodymium/tests/BrowserStatementTest.java

```diff
@@ -96,11 +106,78 @@ public void setJUnitViewModeFlat()
     }
 
     @Test
-    public void testEmptyBrowser()
+    public void testRandomBrowserClassLevel()
     {
         // an empty browser tag (@Browser({""})) should raise an error
-        Result result = JUnitCore.runClasses(EmptyBrowser.class);
-        checkFail(result, 1, 0, 1, "java.lang.IllegalArgumentException: Can not find browser configuration with tag: ");
+        Result result = JUnitCore.runClasses(RandomBrowserClassLevel.class);
+        checkPass(result, 2, 0);
+    }
+
+    @Test
+    public void testRandomBrowserMethodLevel()
+    {
+        // an empty browser tag (@Browser({""})) should raise an error
+        Result result = JUnitCore.runClasses(RandomBrowserMethodLevel.class);
+        checkPass(result, 2, 0);
+    }
+
+    @Test
+    public void testMethodBrowserAnnotationOverwritesClassRandomBrowser()
+    {
+        Result result = JUnitCore.runClasses(MethodBrowserAnnotationOverwritesClassRandomBrowser.class);
+        checkPass(result, 3, 0);
+    }
+
+    @Test
+    public void testRandomBrowserMixed()
+    {
+        Result result = JUnitCore.runClasses(RandomBrowserMixed.class);
+        checkPass(result, 2, 0);
+    }
+
+    @Test
+    public void testRandomBrowsersInheritance()
+    {
+        // the test from RandomBrowserChild should be run 2 times, as the corresponding annotations should be inherited
+        // from the RandomBrowserParent class
+        Result result = JUnitCore.runClasses(RandomBrowsersChild.class);
+        checkPass(result, 2, 0);
+    }
+
+    @Test
+    public void testRandomBrowsersOverwriting()
+    {
+        // the test from RandomBrowsersOverwritingChild should be run 3 times, as the corresponding annotations from
+        // RandomBrowserParent class should be overwritten
+        Result result = JUnitCore.runClasses(RandomBrowsersOverwritingChild.class);
+        checkPass(result, 3, 0);
+    }
+
+    @Test
+    public void testFindBrowserRelatedClassAnnotationMethod()
```

Comment:
This test method is producing an error:
```
java.lang.AssertionError: BrowserStatement.findBrowserRelatedClassAnnotation works not as expected expected:<3> but was:<[@com.xceptance.neodymium.module.statement.browser.multibrowser.RandomBrowsers(value=3)]>
```
The result of the function you're calling is a list of annotations. I guess you want to validate the value instead. Pleas adjust this.

Reply:
Fixed